### PR TITLE
feat(rust) dont store invalid credentials

### LINKED ISSUE #4380: When running the `credential store` command, validate the credential before storing it
## Current

ref: https://github.com/build-trust/ockam/pull/4360/files#r1122971375

When running the `credential store` command, it will store the passed credential associated with the passed issuer. This credential is not validated, so a user could store any string as a valid credential.

## Desired

Verify the credential against the issuer before storing it as it's done at [`credential verify` command](implementations/rust/ockam/ockam_command/src/credential/verify.rs).

---

_We love helping new contributors!_
_If you have questions or need help as you work on your first Ockam contribution, please leave a comment on [this discussion](https://github.com/ockam-network/ockam/discussions/1081)._
_If you're looking for other issues to contribute to, checkout [this discussion](https://github.com/build-trust/ockam/discussions/3206) and labels -  https://github.com/build-trust/ockam/labels/good%20first%20issue or https://github.com/build-trust/ockam/labels/help%20wanted_

### PR DESCRIPTION
**Pull request by crajcan**
## Current behavior

`ockam credential store` will store any string provided.

## Proposed changes
Verify just as `ockam credential verify` does.

https://github.com/build-trust/ockam/issues/4380

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will mark the commit as verified.

### CODE REVIEW

File: implementations/rust/ockam/ockam_command/src/credential/store.rs
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/credential/store.rs:96]: There are some simplifications you can do here. Note that you are returning a tuple with an ok/err pair. You could change that and return a plain Result<String> with the credential.

```rust
let issuer = match &cmd.identity().await {
    Ok(i) => i,
    Err(_) => {
        *is_finished.lock().await = true;
        return Err(miette!("Issuer is invalid").into());
    }
}
.identifier();
if validate_encoded_cred(&cred_as_str, &issuer, &vault_name, &opts)
    .await
    .is_err()
{
    *is_finished.lock().await = true;
    return Err(miette!("Credential is invalid").into());
};
Ok(cred_as_str)
```

File: implementations/rust/ockam/ockam_command/src/credential/store.rs
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/credential/store.rs]: If you apply the suggestion from the other comment, then this part also gets simplified:

```rust
let (credential, _) = try_join!(send_req, progress_output)?;
opts.terminal
    .stdout()
...
```

File: implementations/rust/ockam/ockam_command/tests/bats/credentials.bats
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/tests/bats/credentials.bats:47]: Love it, thanks for adding this test!

File: implementations/rust/ockam/ockam_command/tests/bats/credentials.bats
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/tests/bats/credentials.bats]: Make sure to run the `shfmt` tool to fix any formatting issues. [Check this out](https://github.com/build-trust/ockam/blob/a83c8e9c8dc7b90221dcc0b4cacd45356bda9f98/implementations/rust/ockam/ockam_command/tests/bats/BATS.md#how-to-format-the-tests-scripts).

**Review by adrianbenavides — COMMENTED** [review]
Thank you so much @crajcan for submitting this PR, it looks awesome! I've left some minor comments with a couple of thins to fix.

**Comment by mrinalwadhwa** [comment]
@crajcan Thank you for spending time on this, and an awesome first PR. Before we can merge please

- [ ] Accept the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding your Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will make the commit as verified.

**Comment by davide-baldo** [comment]
Thank you for taking the time to contribute to our project. Your work is valuable, and we would love to see it through to completion. How can we help you in finishing it?

**Comment by crajcan** [comment]
> Thank you for taking the time to contribute to our project. Your work is valuable, and we would love to see it through to completion. How can we help you in finishing it?

My fault, I thought it was just awaiting approval. I just pushed an attempt to fix the lints, I can look at the failing test this afternoon.

**Review by davide-baldo — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**
